### ISSUE TITLE
Not working between mega arduinos

### ISSUE THREAD

**Issue by Emmanuel8107** [victorhurdugaci/SerialProtocol#2]
Hello Victor. As I said in my private mail to you the code does'nt work between 2 mega arduinos. I am using Serial3 to connect them on each board.
I had to redo my own protocol and had similar issue until I discovered that the process time in a loop is so much higher than the Serial frequency that when asking for a Serial3.read the result is often irevelant/wrong : the read has not yet received all the bytes. So I corrected the issue by checking when read gives sufficient bytes to read.
I'll test a solution on your protocol during the coming days and will post my feedback the same time I'll answer your request.
Regards and thank you for the job
Emmanuel

**Comment by kasperv79** [comment]
Hi. 
Is this issue fixed?

**Comment by SoorajR-ai** [comment]
@kasperv79 
no.. only serial 0, 1 and 2 is working.. serial 3 is still not working

**Comment by Emmanuel8107** [comment]
Well I don't remember. Never mind you can close the thread.
Best regards
________________________________________________________________________________
Emmanuel ROBERT
tel : +33 6 50 81 55 21
